src/main/java/racing/CarRace.java

```diff
@@ -1,37 +1,24 @@
 package racing;
 
 import java.util.List;
-import java.util.Random;
+import java.util.stream.Collectors;
 
 public class CarRace {
-    public static final int RANDOM_BOUND = 10;
-    public static final int MOVE_THRESHOLD = 4;
-    private Random randomGenerator;
-    private List<Car> cars;
+    private final List<Car> cars;
 
-    public CarRace(List<Car> cars, Random randomGenerator) {
+    public CarRace(List<Car> cars) {
         if (cars.isEmpty()) {
             throw new IllegalArgumentException("car is empty");
         }
         this.cars = cars;
-        this.randomGenerator = randomGenerator;
-    }
-
-    public CarRace(List<Car> cars) {
-        this(cars, new Random());
     }
 
-    public void race() {
-        cars.forEach(this::race);
-    }
-
-    private void race(Car car) {
-        if (shouldMove()) {
-            car.move();
-        }
+    public void round() {
+        cars.forEach(Car::tryMove);
     }
 
-    private boolean shouldMove() {
-        return randomGenerator.nextInt(RANDOM_BOUND) >= MOVE_THRESHOLD;
+    public List<Car> getWinners() {
+        int maxMoves = cars.stream().mapToInt(Car::getMoves).max().orElseThrow();
```

Comment:
객체지향 생활 체조 원칙의 `규칙 4: 한 줄에 점을 하나만 찍는다.`를 적용하면 가독성이 좋아질 것 같아요.
```suggestion
        int maxMoves = cars.stream()
            .mapToInt(Car::getMoves)
            .max()
            .orElseThrow();
```